File: app/src/main/java/com/example/myscope/activities/Splash_Activity.java
package com.example.myscope.activities;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import com.example.myscope.R;


public class Splash_Activity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.splash_main);

        new Handler().postDelayed(new Runnable() {

// Using handler with postDelayed called runnable run method

            @Override

            public void run() {

                Intent i = new Intent(Splash_Activity.this, Login_Page.class);

                startActivity(i);

                // close this activity

                finish();

            }

        }, 4000);

    }
}

File: app/src/main/java/com/example/myscope/activities/Splash_Activity.kt
package com.soargtechnologies.myscope.activities

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import com.soargtechnologies.myscope.R

class Splash_Activity : BaseActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.splash_main)
        Handler().postDelayed({
            navigateToActivity(Intent(applicationContext,Login_Page::class.java))
            finish()
        }, 2000)
    }
}